### PR TITLE
#160345747 Enable User Logout

### PR DESCRIPTION
**Pull request by munyili-samuel**
#### What does this PR do?
- adds a function to handle user logout request event

#### Description of Task to be completed?
- added a logout function to implement logout on Redux and clear user data from localStorage
- added `history prop` to the `UserNavBar` to handle browser history
- added a `handleLogout` function to handle the request when the user clicks the logout link

#### How should this be manually tested?
- use git to pull for the branch `ft-user-logout-160345747`
- update your environmental variables on the `.env` file. Be sure to set a value for the `REACT_APP_API_URL` to point to the back-end server
- run `source .env` to export the env variables
- run `npm install` to update dependencies
- run `npm start -s` to start the server
- use the URL `/login' to log in a user
- once logged in copy the URL to the dashboard and paste in a different tab on the same browser window. The dashboard should still be accessible.
- on the right end of the blue navigation bar, click on the drop-down link next to the username then click `Sign Out`
- The URL should change to `/login` and you will be taken to the login window.
- If you try accessing the `/dashboard` URL you copied earlier, you will be taken back to the login window

**Note:** If you are using a local server for the backend and experience issues receiving responses, you might consider using a project hosted on `heroku`

#### What are the relevant pivotal tracker stories?
- [#160345747](https://www.pivotaltracker.com/story/show/160345747)

### CODE REVIEW

File: .env-sample [hunk 1]
```diff
@@ -10,3 +10,6 @@ REACT_APP_FACEBOOK_APP_SECRET='facebook-app-secret-here''
 
 # The url for social login in our backend
 REACT_APP_SOCIAL_URL='url-here
+
+# The ur for user login
```
Comment (RuthNjeri) [inline .env-sample]: Hey Sam,

You have misspelt url in the comment.

**Review by RuthNjeri — DISMISSED** [review]

File: .env-sample (hunk 1 above)
Comment (hoslack) [inline .env-sample]: Please name the API_URL to something more descriptive

File: src/actions/auth.js [hunk 2]
```diff
@@ -1,10 +1,22 @@
-import { USER_LOGGED_IN } from './types';
+import { USER_LOGGED_IN, SIGN_OUT_DONE } from './types';
```
Comment (hoslack) [inline src/actions/auth.js]: Please import each type on its own line

**Review by hoslack — CHANGES_REQUESTED** [review]

File: .env-sample (hunk 1 above)
Comment (munyili-samuel) [reply .env-sample]: @RuthNjeri, I have corrected that

Comment (munyili-samuel) [reply .env-sample]: @hoslack I have corrected that

**Review by lennykioko — APPROVED** [review]
Has been tested and works as expected.

**Review by davenjeru — APPROVED** [review]
Tested it with @lennykioko and it works as expected. On logout, it deletes user information from local storage and redirects to the root path.

**Review by hoslack — APPROVED** [review]

**Pull request merged**